lib/driver.js

```diff
@@ -22,13 +21,20 @@ class AndroidDriver extends BaseDriver {
     [sessionId] = await super.createSession(caps);
 
     // TODO fail if app file is missing, or if app is not on the device?
-
+    // assigning defaults
+    let defaultOpts = {action: "android.intent.action.MAIN",
+                       category: "android.intent.category.LAUNCHER",
+                       flags: "0x10200000",
+                       disableAndroidWatchers: false};
+    _.defaults(this.opts, defaultOpts);
     if (!this.opts.javaVersion) {
       this.opts.javaVersion = await helpers.getJavaVersion();
     }
-
+    // Add condition for chrome driver
+    await this.checkAppPresent();
```

Comment:
i think it would make more sense to call `configureApp` here before we set up lot of things. As if configure app fails that would mean we have malformed url or incorrect zip file or incorrect apk. And hence we can save lot of time that would be other wise wasted in initializing device